op-chain-ops/interopgen/recipe.go

```diff
@@ -0,0 +1,280 @@
+package interopgen
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/params"
+
+	"github.com/ethereum-optimism/optimism/op-chain-ops/devkeys"
+	"github.com/ethereum-optimism/optimism/op-chain-ops/genesis"
+)
+
+type InteropDevRecipe struct {
+	L1ChainID        uint64
+	L2ChainIDs       []uint64
+	GenesisTimestamp uint64
+}
+
+func (r *InteropDevRecipe) Build(addrs devkeys.Addresses) (*WorldConfig, error) {
+	// L1 genesis
+	l1Cfg := &L1Config{
+		ChainID: new(big.Int).SetUint64(r.L1ChainID),
+		DevL1DeployConfig: genesis.DevL1DeployConfig{
+			L1BlockTime:             6,
+			L1GenesisBlockTimestamp: hexutil.Uint64(r.GenesisTimestamp),
+			L1GenesisBlockGasLimit:  30_000_000,
+		},
+		Prefund: make(map[common.Address]*big.Int),
+	}
+
+	l1Users := devkeys.ChainUserKeys(l1Cfg.ChainID)
+	for i := uint64(0); i < 20; i++ {
```

Comment:
FYI I cut this because in `SuperSystem` there's this weird implicit "20 accounts are funded" logic. You could make it a `TODO` if you wished:
https://github.com/ethereum-optimism/optimism/issues/11887